src/app.rs

```diff
@@ -378,257 +408,362 @@ impl UrdfViewerApp {
-                    if ro.requested {
-                        match self.set_robot_origin_from_request(&ro.origin) {
-                            Ok(_) => {
-                                self.update_robot();
-                                ro.requested = false;
-                            }
-                            Err(err) => {
-                                println!("{}", err);
-                            }
+                        Err(err) => {
+                            println!("{}", err);
                         }
                     }
                 }
-                if let Ok(mut cur_ro) = data.current_robot_origin.lock() {
-                    let o = self.robot.origin();
-                    for i in 0..3 {
-                        cur_ro.position[i] = o.translation.vector[i];
-                    }
-                    cur_ro.quaternion[0] = o.rotation.quaternion().w;
-                    cur_ro.quaternion[1] = o.rotation.quaternion().i;
-                    cur_ro.quaternion[2] = o.rotation.quaternion().j;
-                    cur_ro.quaternion[3] = o.rotation.quaternion().k;
-                }
             }
-            if self.has_arms() {
-                let name = &self
-                    .get_arm()
-                    .iter()
-                    .last()
-                    .unwrap()
-                    .joint()
-                    .name
-                    .to_owned();
-                self.viewer.draw_text(
-                    &format!("IK target name [{}]", name),
-                    FONT_SIZE_INFO,
-                    &na::Point2::new(10f32, 100.0),
-                    &na::Point3::new(0.5f32, 0.8, 0.2),
-                );
+            #[cfg(not(target_arch = "wasm32"))]
+            if let Ok(mut cur_ja) = self.data.current_joint_positions.lock() {
+                cur_ja.positions = self.app.robot.joint_positions();
             }
-            if is_ctrl && !is_shift {
-                self.viewer.draw_text(
-                    "moving joint by drag",
-                    FONT_SIZE_INFO,
-                    &na::Point2::new(10f32, 150.0),
-                    &na::Point3::new(0.9f32, 0.5, 1.0),
-                );
+
+            // Robot orientation for web server
+            #[cfg(not(target_arch = "wasm32"))]
+            if let Ok(mut ro) = self.data.target_robot_origin.lock() {
+                if ro.requested {
+                    self.app.set_robot_origin_from_request(&ro.origin);
+                    self.app.update_robot();
+                    ro.requested = false;
+                }
             }
-            if is_shift {
-                self.viewer.draw_text(
-                    "solving ik",
-                    FONT_SIZE_INFO,
-                    &na::Point2::new(10f32, 150.0),
-                    &na::Point3::new(0.9f32, 0.5, 1.0),
-                );
+            #[cfg(not(target_arch = "wasm32"))]
```

Comment:
`#[cfg(not(target_arch = "wasm32"))]` is repeated four times.
Is it possible to reduce it to make it easy to understand the logic?
Adding a (member) function (`handle_http_request()`) to do these?

Reply:
I think if you have a future plan to handle these for wasm, then you can skip it for now, and do it in that timing.

Reply:
Extracted its logic to handle_http_request method.